src/coreComponents/constitutive/fluid/multifluid/MultiFluidSelector.hpp

```diff
@@ -79,6 +79,71 @@ void constitutiveUpdatePassThru( MultiFluidBase & fluid,
                                >::execute( fluid, std::forward< LAMBDA >( lambda ) );
 }
 
+namespace detail
+{
+/**
+ * @brief Structure to execute a lambda on a fluid model depending on the number of components
+ * @tparam THERMAL_ACTIVE Determines if the call has been deactivated due to the lambda being thermal
+ *         but the fluid model not being thermal
+ * @tparam COMPONENT_LIST Type listing the components to expand over
+ */
+template< bool THERMAL_ACTIVE, typename COMPONENT_LIST >
+struct ComponentSelector;
+
+/** @brief If the call is deactivated due to the fluid not being thermal
+ */
+template< typename COMPONENT_LIST >
+struct ComponentSelector< false, COMPONENT_LIST >
```

Comment:
instead of specializing the implementation you can `if constexpr` that boolean template parameter and have a single implementation. I let you decide where to put the `if constexpr`. It could even go directly inside `constitutiveComponentUpdatePassThru` and do something like:

````c++
if constexpr (active)
{
detail::ComponentSelector< active, Components >::execute( numComps, fluid, std::forward< LAMBDA >( lambda ));
} 
else
{
/// throw an error. 
}